src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -5701,6 +5701,60 @@ CancellationCode StorageReplicatedMergeTree::killMutation(const String & mutatio
     return CancellationCode::CancelSent;
 }
 
+void StorageReplicatedMergeTree::removePartsFromFilesystem(const DataPartsVector & parts)
+{
+    const auto settings = getSettings();
+    if (settings->max_part_removal_threads > 1 && parts.size() > settings->concurrent_part_removal_threshold)
+    {
+        /// Parallel parts removal.
+
+        size_t num_threads = std::min(size_t(settings->max_part_removal_threads), parts.size());
```

Comment:
Consider to use static_cast instead of c_style cast.

Reply:
Done by explicitly passing the type to the std::min